package ServerMessage;

message ServerInformation {
 
  required int32 serverInformation = 1;
  required int64 messageNumber = 2;
  
}

message playerPos{
    required float posY = 1;
    required float posX = 2;

}

message Playerinfromation {

  required int32 playerNumber = 1;
  required playerPos pos = 3;
  required int32 playerScore = 4; 
}



message AdditionalInformation{


}   
message ServerMessage{
     

     // The player number of the client
    required int32 playersConnected = 1;



    required ServerInformation serverinfo = 2;
    required Playerinfromation playerOne = 3;
    required Playerinfromation playerTwo = 4;
 
     //The client player infromation
    required int32 playerNumber = 5;

    required AdditionalInformation additioanlInfo = 8;
    /*
Number of players within a game
*/
const int NUM_PLAYERS = 2;

}


